src/webgpu/api/validation/encoding/cmds/index_access.spec.ts

```diff
@@ -0,0 +1,167 @@
+      glsl: `
+        #version 310 es
+        precision mediump float;
+        layout(location = 0) out vec4 fragColor;
+        void main() {
+            fragColor = vec4(0.0, 1.0, 0.0, 1.0);
+        }
+      `,
+    });
+
+    return this.device.createRenderPipeline({
+      layout: this.device.createPipelineLayout({ bindGroupLayouts: [] }),
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      primitiveTopology: 'triangle-strip',
+      colorStates: [{ format: 'rgba8unorm' }],
+      vertexState: {
+        vertexBuffers: [
+          {
+            arrayStride: 4 * 4,
+            stepMode: 'vertex',
+            attributes: [{ format: 'float4', offset: 0, shaderLocation: 0 }],
+          },
+        ],
+      },
+    });
+  }
+
+  beginRenderPass(encoder: GPUCommandEncoder) {
+    const colorAttachment = this.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 100, height: 100, depth: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    return encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: colorAttachment.createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+          storeOp: 'store',
+        },
+      ],
+    });
+  }
+
+  drawIndexed(
+    indexCount: number,
+    instanceCount: number,
+    firstIndex: number,
+    baseVertex: number,
+    firstInstance: number
+  ) {
+    const [vertexBuffer, indexBuffer] = this.createVertexAndIndexBuffer();
+
+    const pipeline = this.createRenderPipeline();
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = this.beginRenderPass(encoder);
+    pass.setPipeline(pipeline);
+    pass.setVertexBuffer(0, vertexBuffer);
+    pass.setIndexBuffer(indexBuffer);
+    pass.drawIndexed(indexCount, instanceCount, firstIndex, baseVertex, firstInstance);
+    pass.endPass();
+
+    this.device.defaultQueue.submit([encoder.finish()]);
+  }
+
+  drawIndexedIndirect(bufferArray: Uint32Array, indirectOffset: number) {
+    const [indirectBuffer, indirectMapping] = this.device.createBufferMapped({
+      size: bufferArray.byteLength,
+      usage: GPUBufferUsage.INDIRECT,
+    });
+    new Uint32Array(indirectMapping).set(bufferArray);
+    indirectBuffer.unmap();
+
+    const [vertexBuffer, indexBuffer] = this.createVertexAndIndexBuffer();
+
+    const pipeline = this.createRenderPipeline();
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = this.beginRenderPass(encoder);
+    pass.setPipeline(pipeline);
+    pass.setVertexBuffer(0, vertexBuffer);
+    pass.setIndexBuffer(indexBuffer, 0);
+    pass.drawIndexedIndirect(indirectBuffer, indirectOffset);
+    pass.endPass();
+
+    this.device.defaultQueue.submit([encoder.finish()]);
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('draw_indexed_index_access_out_of_bounds').fn(t => {
```

Comment:
Can be parameterized to make it easier to add additional cases, perhaps something like this:

```ts
g.test('out_of_bounds')
  .params(
    params()
      .combine(pbool('indirect'))
      .combine([
        { indexCount: 7, firstIndex: 0 },
        { indexCount: 10000, firstIndex: 0 },
        { indexCount: 6, firstIndex: 1 },
        { indexCount: 6, firstIndex: 10000 },
      ] as const)
      .combine(poptions('instanceCount', [1, 10000]))
  )
  .fn(t => {
    const { indirect, indexCount, firstIndex, instanceCount } = t.params;
    if (indirect) {
      t.drawIndexedIndirect(new Uint32Array([indexCount, instanceCount, firstIndex, 0, 0), 0);
    } else {
      t.drawIndexed(indexCount, instanceCount, firstIndex, 0, 0);
    }
  });
```